src/main/java/com/google/research/bleth/utils/Queries.java

```diff
@@ -116,4 +117,25 @@ public static double average(List<Entity> entities, String property) throws Clas
         return properties.stream()
                 .collect(toImmutableMap(Function.identity(), property -> average(entities, property)));
     }
+
+    /**
+     * Given a simulationId as a string, delete all data associated with that simulation id.
+     * @param simulationId is the simulation id.
+     */
+    public static void delete(String simulationId) {
+        delete(Schema.StatisticsState.entityKindBeaconsObservedPercent, Schema.StatisticsState.simulationId, simulationId);
+        delete(Schema.StatisticsState.entityKindDistance, Schema.StatisticsState.simulationId, simulationId);
+        delete(Schema.BoardState.entityKindReal, Schema.BoardState.simulationId, simulationId);
+        delete(Schema.BoardState.entityKindEstimated, Schema.BoardState.simulationId, simulationId);
+        datastore.delete(KeyFactory.stringToKey(simulationId));
+    }
+
+    private static void delete(String entityKind, String foreignKeyProperty, String foreignKeyKeyValue) {
+        Query.Filter filter = new Query.FilterPredicate(foreignKeyProperty, Query.FilterOperator.EQUAL, foreignKeyKeyValue);
+        Query entitiesToDelete = new Query(entityKind).setFilter(filter);
+        Iterable<Entity> entitiesToDeleteIterable = datastore.prepare(entitiesToDelete).asIterable();
+        for (Entity entity : entitiesToDeleteIterable) {
```

Comment:
This loop is not strictly needed: `DataStoreService` has a `delete` method accepting an `Iterable` directly. (You can transform the `Iterable<Entity>` to an `Iterable<Key>` using Guava `Iterables.transform()`)